src/core/operations/ParseX509CertificateBundles.mjs

```diff
@@ -0,0 +1,150 @@
+/**
+ * @author nehagopinath [nehagowri94@gmail.com]
+ * @copyright Crown Copyright 2020
+ * @license Apache-2.0
+ */
+
+import Operation from "../Operation.mjs";
+import r from "jsrsasign";
+import {formatDnStr} from "../lib/PublicKey";
+
+/**
+ * ParseX509CertificateBundles operation
+ */
+class ParseX509CertificateBundles extends Operation {
+
+    /**
+     * ParseX509CertificateBundles constructor
+     */
+    constructor() {
+        super();
+
+        this.name = "ParseX509CertificateBundles";
+        this.module = "PublicKey";
+        this.description = "X.509 is an ITU-T standard for a public key infrastructure (PKI) and Privilege Management Infrastructure (PMI). It is commonly involved with SSL/TLS security.<br><br>This operation displays the required contents of different certificates bundled in the input file in a human readable format, similar to the openssl command line tool.<br><br>Tags: X509, server hello, handshake";
+        this.infoURL = "https://wikipedia.org/wiki/X.509";
+        this.inputType = "string";
+        this.outputType = "string";
+        this.args = [
+            {
+                "name": "Input format",
+                "type": "option",
+                "value": ["PEM"]
+            }
+        ];
+        this.patterns = [
+            {
+                "match": "^-+BEGIN CERTIFICATE-+\\r?\\n[\\da-z+/\\n\\r]+-+END CERTIFICATE-+\\r?\\n?$",
+                "flags": "i",
+                "args": [
+                    "PEM"
+                ]
+            }
+        ];
+    }
+
+    /**
+     * @param {string} input
+     * @param {Object[]} args
+     * @returns {string}
+     */
+    run(input, args) {
+
+        if (!input.length) {
+            return "No input";
+        }
+
+        const regex1 = /^-----BEGIN CERTIFICATE-----\r?\n((?:(?!-----).*\r?\n)*)-----END CERTIFICATE-----/gm;
+        const regex2 = /^-----BEGIN TRUSTED CERTIFICATE-----\r?\n((?:(?!-----).*\r?\n)*)-----END TRUSTED CERTIFICATE-----/gm;
+
+        let m;
+        let res = "";
+        let count = 0;
+
+        while ((m = regex1.exec(input)) !== null) {
+            // This is necessary to avoid infinite loops with zero-width matches
+            if (m.index === regex1.lastIndex) {
+                regex1.lastIndex++;
+
+            }
+
+            count++;
+            res += "\nCertificate:\n" + parseCert("-----BEGIN CERTIFICATE-----" + "\n" + m[1] + "\n" + "-----END CERTIFICATE-----");
```

Comment:
Use template strings to tidy this up a little. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals